setup.sh

```diff
@@ -14,4 +14,31 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-pip3 install -r requirements.txt -f https://storage.googleapis.com/jax-releases/libtpu_releases.html
\ No newline at end of file
+# Uninstall existing JAX and JAXlib
+pip3 show jax && pip3 uninstall -y jax 
+pip3 show jaxlib && pip3 uninstall -y jaxlib
+
+# Install dependencies from requirements.txt
+pip3 install -r requirements.txt -f https://storage.googleapis.com/jax-releases/libtpu_releases.html
+
+# Set environment variables
+for ARGUMENT in "$@"; do
+    IFS='=' read -r KEY VALUE <<< "$ARGUMENT"
+    export "$KEY"="$VALUE"
+done
+
+if [[ $MODE == "nightly" ]]; then
+    echo "Installing jax-head, jaxlib-nightly, and libtpu-nightly"
+    # Install JAX from GitHub head
+    pip3 install git+https://github.com/google/jax
+    # Install jaxlib-nightly
+    pip3 install --pre -U jaxlib -f https://storage.googleapis.com/jax-releases/jaxlib_nightly_releases.html
+    # Install libtpu-nightly
+    pip3 install libtpu-nightly -f https://storage.googleapis.com/jax-releases/libtpu_releases.html -U --pre
+fi
+
+if [[ -n "$LIBTPU_GCS_PATH" ]]; then
+    sudo pip3 install -U crcmod
+    echo "Copying libtpu.so from $LIBTPU_GCS_PATH to /lib/libtpu.so"
+    sudo gsutil cp "$LIBTPU_GCS_PATH" /lib/libtpu.so
+fi
```

Comment:
Having now read to this point, I recommend the following order.

(1) Uninstall jax, jaxlib and libtpu and delete `/lib/libtpu.so`
(2) Install jax, jaxlib and libtpu
- if `stable`, via jax[tpu]. Assert the user doesn't pass in a libtpu separately.
- if `nightly`, install jax and jaxlib as you do. Install libtpu from either GCS or libtpu nightly as appropriate. 
(3) install the rest of requirements.

Reply:
Just made the change based on your comments.